test/parallel/test-process-env-allowed-flags.js

```diff
@@ -62,15 +62,38 @@ const assert = require('assert');
                      true);
 
   process.allowedNodeEnvironmentFlags.add('foo');
+  Set.prototype.add.call(process.allowedNodeEnvironmentFlags, 'foo');
   assert.strictEqual(process.allowedNodeEnvironmentFlags.has('foo'), false);
-  process.allowedNodeEnvironmentFlags.forEach((flag) => {
-    assert.strictEqual(flag === 'foo', false);
-  });
+
+  const thisArg = {};
+  process.allowedNodeEnvironmentFlags.forEach(
+    common.mustCallAtLeast(function(flag, _, set) {
+      assert.notStrictEqual(flag, 'foo');
+      assert.strictEqual(this, thisArg);
+      assert.strictEqual(set, process.allowedNodeEnvironmentFlags);
+    }),
+    thisArg
+  );
+
+  for (const flag of process.allowedNodeEnvironmentFlags.keys()) {
+    assert.notStrictEqual(flag, 'foo');
+  }
+  for (const flag of process.allowedNodeEnvironmentFlags.values()) {
+    assert.notStrictEqual(flag, 'foo');
+  }
+  for (const flag of process.allowedNodeEnvironmentFlags) {
+    assert.notStrictEqual(flag, 'foo');
+  }
+  for (const [flag] of process.allowedNodeEnvironmentFlags.entries()) {
+    assert.notStrictEqual(flag, 'foo');
+  }
 
   process.allowedNodeEnvironmentFlags.clear();
+  Set.prototype.clear.call(process.allowedNodeEnvironmentFlags);
   assert.strictEqual(process.allowedNodeEnvironmentFlags.size > 0, true);
 
   const size = process.allowedNodeEnvironmentFlags.size;
   process.allowedNodeEnvironmentFlags.delete('-r');
+  Set.prototype.delete.call(process.allowedNodeEnvironmentFlags, '-r');
   assert.strictEqual(process.allowedNodeEnvironmentFlags.size, size);
```

Comment:
Optional nit: Maybe keep an assertion after each call so it's very clear which call isn't working as expected?

Also, while we're at it, maybe some general cleanup and tightening of the assertions?

```suggestion
  const size = process.allowedNodeEnvironmentFlags.size;

  process.allowedNodeEnvironmentFlags.clear();
  assert.strictEqual(process.allowedNodeEnvironmentFlags.size, size);
  Set.prototype.clear.call(process.allowedNodeEnvironmentFlags);
  assert.strictEqual(process.allowedNodeEnvironmentFlags.size, size);
  
  process.allowedNodeEnvironmentFlags.delete('-r');
  assert.strictEqual(process.allowedNodeEnvironmentFlags.size, size);
  Set.prototype.delete.call(process.allowedNodeEnvironmentFlags, '-r');
  assert.strictEqual(process.allowedNodeEnvironmentFlags.size, size);
```